Add disconnect feature to api

abdulkadirozyurt opened the pull request:
Hi!
Sorry about that. I had to delete my fork. 
I created a new pull request for this.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added new publisher disconnect endpoint (DELETE /api/disconnect/{publisher_id}) with admin/write permissions required.

* **Documentation**
  * Updated API documentation with disconnect endpoint details, rate limiting, and usage examples.

* **Chores**
  * Updated project dependencies and ignore patterns.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

abdulkadirozyurt committed:
feat(docker): Add spdlog to runtime dependencies

abdulkadirozyurt committed:
feat(SLSManager): Add disconnect_publisher method to manage publisher disconnections

abdulkadirozyurt committed:
feat(API): Add disconnect publisher endpoint and update documentation

abdulkadirozyurt committed:
Delete test_disconnect_api.md

abdulkadirozyurt committed:
SLSApiServer.cpp: Add publisher disconnect API endpoint

- Implemented DELETE /api/disconnect/{publisher_id} endpoint for disconnecting active publishing operations by publisher ID.
- Updated handler to use publisher_id directly instead of player_id.
- Improved permission checks and error handling for disconnect operation.
- Refactored code for clarity and RESTful compliance.

abdulkadirozyurt committed:
SLSManager.hpp: Refactor disconnect and stats API to use publisher_id

- Remove all player_id and player-to-publisher mapping logic from the interface.
- Update disconnect_publisher and generate_json_for_publisher signatures to use publisher_id directly.
- Remove find_publisher_by_player_key declaration.
- Clarify API for publisher-based operations.

abdulkadirozyurt committed:
SLSManager.cpp: Refactor disconnect logic to use publisher_id

- Remove all player_id and player-to-publisher mapping logic from implementation.
- Update disconnect_publisher and generate_json_for_publisher to use publisher_id directly.
- Remove find_publisher_by_player_key and related code.
- Fix all syntax and brace issues from refactor.
- Ensure all publisher operations are consistent and robust.

abdulkadirozyurt committed:
API.md: Update documentation for publisher_id-based disconnect endpoint

- Document new API endpoint for disconnecting publishers using publisher_id.
- Remove references to player_id and player-to-publisher mapping.
- Clarify usage and expected parameters for disconnect operation.
- Ensure documentation matches refactored backend logic.

servusrene commented on slscore/SLSManager.cpp:
```diff
@@ -282,12 +239,35 @@ json CSLSManager::generate_json_for_publisher(std::string playerKey, int clear,
     return ret;
 }
 
+bool CSLSManager::disconnect_publisher(const std::string& player_key) {
```
the varibale should be named as publisher_key

servusrene commented on slscore/SLSManager.cpp:
```diff
@@ -195,62 +195,19 @@ int CSLSManager::start()
     }
 
-    // Validate player key and get mapped publisher key
-    char* mapped_publisher = find_publisher_by_player_key(const_cast<char*>(playerKey.c_str()));
```
we need the player to receive our stats, so why have you deleted the code?

servusrene commented on slscore/SLSManager.cpp:
```diff
@@ -195,62 +195,19 @@ int CSLSManager::start()
 
 }
 
-char* CSLSManager::find_publisher_by_player_key(char *player_key) {
```
same as in line 244: we need the player to receive our stats, so why have you deleted the code?

servusrene reviewed (changes requested):
It looks quite good already, there are still some minor inconsistencies in the code.

abdulkadirozyurt committed:
refactor(SLSManager): update disconnect logic to use publisher_key

- Rename parameter from player_key to publisher_key in disconnect_publisher method.
- Update logging messages to reflect the change from player_key to publisher_key.

abdulkadirozyurt committed:
Improve config parsing robustness: handle whitespace, comments, and line endings for cross-platform compatibility

- Enhance trim and replace logic to ensure config files are parsed correctly regardless of OS or editor
- Improve error logging and block parsing for better maintainability
- Prevent parsing errors caused by stray whitespace, tabs, or comment lines
- Ensures reliable config loading in Docker and all deployment environments

abdulkadirozyurt committed:
feat(SLSManager): enhance publisher disconnection logic
- Update disconnect logic to support both publisher ID and player ID.
- Refactor generate_json_for_publisher to accept player key instead of publisher ID.
- Add find_publisher_by_player_key method to resolve player keys to publisher IDs.
- Improve logging for publisher disconnection and JSON generation processes.

abdulkadirozyurt commented:
@servusrene Some code was unintentionally removed in the previous changes. I’ve restored it so players can continue receiving stats and corrected the variable name in disconnect_publisher.

abdulkadirozyurt commented:
Hi @servusrene,

I’ve addressed the review feedback in PR #4:
- Renamed variables for consistency (e.g., publisher_key naming)
- Restored the player_key -> publisher mapping / stats flow (so players can still receive stats as before)

The PR is currently shown as mergeable, but it still has a “changes requested” review, so I’m guessing that might be what’s blocking the merge.

Is there anything else you’d like me to adjust before it can be merged?
If everything looks good now, could you please re-review / approve?

Thanks!

servusrene commented:
Hey @abdulkadirozyurt,

Please excuse the long wait. I am currently working behind the scenes on an improved version of the srt-live server. Among other things, the API is being optimized. This means that this PR would not be 100% compatible right away. As soon as the version is released, I will merge this feature with the adjustments for the new API function.

However, I can't say at this point when the new version will be released.

abdulkadirozyurt commented:
Her @servusrene 
Thanks for the clarification — I understand the reason you’re not merging this PR at the moment.
I appreciate you taking the time to review and explain the decision.
If anything changes in the future or there’s a preferred direction for implementing this, I’d be happy to help.
